iceoryx2/src/config.rs

```diff
@@ -69,22 +69,20 @@
 //! # }
 //! ```
 
+use dirs;
```

Comment:
Rather than using this directly, an abstraction should be implemented.

This allows for swapping out the implementation and will make it easier to isolate for safety certification purposes. I haven't looked at the implementation of `dirs` closely, but it might not be suitable for certification e.g. if it uses heap.

The cleanest way would probably be to create a "concept" in the concept abstraction layer (`iceoryx2-cal`) for configuration storage, then add one implementation that uses `dirs`.

This will allow for:
1. Swapping out an implementation to a safety-certifiable version
2. Implementation of the concept for arbitrary environments that don't have a file system

cc. @elfenpiff @elBoberido